src/main/kotlin/domain/usecase/GetIraqiMealsUseCase.kt

```diff
@@ -0,0 +1,26 @@
+package org.example.domain.usecase
+
+import org.example.domain.FoodException
+import org.example.domain.repository.FoodRepository
+import org.example.model.Food
+
+    class GetIraqiMealsUseCase(private val repository: FoodRepository) {
+
+        fun getIraqiMeals(): List<Food> {
```

Comment:
I think it would be so much better to use 
```suggestion
        fun getIraqiMeals(): Result<List<Food>> {
```

check out GetEasyFoodSuggestionUseCase